lib/src/http.dart

```diff
@@ -63,32 +63,7 @@ class _PubHttpClient extends http.BaseClient {
     request.headers[HttpHeaders.userAgentHeader] = "Dart pub ${sdk.version}";
     _logRequest(request);
 
-    http.StreamedResponse streamedResponse;
-    try {
-      streamedResponse = await _inner.send(request);
-    } on SocketException catch (error, stackTraceOrNull) {
-      // Work around issue 23008.
-      var stackTrace = stackTraceOrNull ?? Chain.current();
-
-      if (error.osError == null) rethrow;
-
-      if (error.osError.errorCode == 8 ||
-          error.osError.errorCode == -2 ||
-          error.osError.errorCode == -5 ||
-          error.osError.errorCode == 11001 ||
-          error.osError.errorCode == 11004) {
```

Comment:
@nex3, you wouldn't happen to know where the error codes come from? and why we don't retry them..

Reply:
I don't remember for sure, but I think they came up from manual testing on what happened when we tried to request obviously bogus or non-existent URLs on various OSes.

Reply:
Thanks, that makes sense... I suspect we can make slight reliability improvements by retrying..

But th tests do reveal that straight up ignoring them produce worse error messages :)